### DISCUSSION TITLE
Github Copilot not working on Pycharm anymore

### DISCUSSION THREAD

**Discussion by mingchau** [community/community#18793] [Copilot Conversations] [answered]
The copilot tool window keeps saying "Please login to Github and try again."
Pycharm version: PyCharm 2022.1.2 (Professional Edition)
Github Copilot plugin version: 1.1.26.1691
curl --noproxy '*' github.com returns nothing

Attached the related log information.
![Tool window message](https://user-images.githubusercontent.com/18564039/174277285-cfae9d98-4ae8-4717-b3d4-d4abcb79950e.png)
![curl --noproxy](https://user-images.githubusercontent.com/18564039/174277320-53357909-5fd6-4a88-9481-d101a04a7498.png)
![Pycharm logs](https://user-images.githubusercontent.com/18564039/174277339-1015db1c-df7f-48c0-a78c-5792863ad502.png)

**Comment by mingchau** [comment, marked as answer]
The problem was solved by going to Pycharm `Tools -> Github Copilot -> Login to Github`.
It's because the login window didn't pop out automatically.

**Reply by urmichm** [reply]
did not help, still the same problem

**Reply by kv0409** [reply]
Worked for me. Thanks!

**Reply by lelandg** [reply]
Thanks, @mingchau. This helped me find the problem today. 

I think it's now under `Tools -> GitHub Copilot -> Settings -> GitHub Copilot Account Settings`. Or if you don't see the latter, `Manage GitHub Account` to get logged in, if you're not. Apparently, you get logged out if you uninstall/reinstall.

**Comment by 3sv1n** [comment]
nope doesn't work for me... I recently signed up as a student and installed pycharm professsional, and even after logging in to copilot, the "waiting for verification" box disappears and nothing happens... have done the entire for about 7 times now... pls help me...

**Reply by davecheney** [reply]
Please see https://github.com/orgs/community/discussions/29127 for HTTP proxy support.

**Reply by 3sv1n** [reply]
@davecheney still doesn't work... I enabled the localhost as specified in the instructions but it doesnt work
the website gets opened and access is granted, but somehow, the pycharm application is not able to update

**Reply by shanielk9** [reply]
Same with me.
any solution ?

**Reply by 3sv1n** [reply]
> Same with me. any solution ?

Hey there! 
Check out the solution that i have posted, you need to start it again in your profile settings after verifying studentship

**Reply by artificial-cesare** [reply]
Hi, were did you write your reply? Struggling with the same problem here...

**Comment by gius28** [comment]
Same here. 
Pycharm 2022.2.1 professional edition + Github Copilot 1.1..31.1908 
Working on normal .py file but not on jupyter

**Reply by 3sv1n** [reply]
Hi my issue was fixed by doing the following:

after verifying as student, you need to renew the copilot license;
profile > settings > billing and payment > copilot 
follow the instructions to renew the license and it works!

**Reply by vmuu** [reply]
配置文件 > 设置 > 计费和付款 > copilot
这个是在哪个位置啊，我搞了两天，人都快傻了

**Reply by someoneB** [reply]
> 配置文件 > 设置 > 计费和付款 > copilot 这个是在哪个位置啊，我搞了两天，人都快傻了

应该在github账户里，但是我没有看到更新许可证，好奇怪

**Comment by gius28** [comment]
Revert back your pycharm version.
It works again! I hope this will be fixed in the next version.

**Comment by ppokrovskii** [comment]
Resolved by going to https://github.com/settings/copilot and completing the setup (had to select to allow or not allow to use public code)

**Reply by MustafaMaged** [reply]
worked for me, thanks a lot!

**Reply by AmineDjeghri** [reply]
this. 
it worked for me !

**Comment by LukasKristensen** [comment]
Had to update the copilot to solve this:
https://plugins.jetbrains.com/plugin/17718-github-copilot

**Reply by urmichm** [reply]
this did not help me :(

**Comment by neoneo40** [comment]
I have solution.
First pay(30 free trial) then again. login success

**Comment by BartlomiejSkwiraAML** [comment]
Updating Pycharm and Copilot plugin solved the issue for me.
